packages/shared/src/lib/tracking.ts

```diff
@@ -70,6 +71,52 @@ export const addRegionIdAndTrackingParamsToSupportUrl = (
         : baseUrl;
 };
 
+// TRACKING VIA profile.theguardian.com
+type ProfileLinkParams = {
+    componentEventParams: string;
+    returnUrl: string;
+};
+
+export const addProfileTrackingParams = (baseUrl: string, params: Tracking): string => {
+    const constructQuery = (query: { [key: string]: any }): string =>
+        Object.keys(query)
+            .map((param: string) => {
+                const value = query[param];
+                const queryValue = Array.isArray(value)
+                    ? value.map(v => encodeURIComponent(v)).join(',')
+                    : encodeURIComponent(value);
+                return `${param}=${queryValue}`;
+            })
+            .join('&');
+
+    const componentEventParamsData = {
```

Comment:
@tomrf1  I'd like to add browser id and visit id here if possible but requires reading a cookie (ideally via @guardian/libs). Any issue with me adding? I noticed the shared package json is quite minimal and it's a client-side specific use-case. Not strictly necessary but nice to have.

Reply:
Build is failing due to the last commit that adds this. It may not be necessary so will delete and revert to previous commit tomorrow if the data analysts confirms the cookie data is not needed.